napari/layers/transforms.py

```diff
@@ -0,0 +1,107 @@
+    """
+
+    def __init__(self, func=tz.identity, inverse=None, name=None):
+        self.func = func
+        self._inverse_func = inverse
+        self.name = name
+
+        if func is tz.identity:
+            self._inverse_func = tz.identity
+
+    def __call__(self, coords):
+        """Transform input coordinates to output."""
+        return self.func(coords)
+
+    def set_slice(self, axes: Sequence[int]):
+        """Return a transform subset to the visible dimensions."""
+        raise NotImplementedError('Cannot subset arbitrary transforms.')
+
+    def set_pad(self, axes: Sequence[int]):
+        """Return a transform with added axes for non-visible dimensions."""
+        raise NotImplementedError('Cannot subset arbitrary transforms.')
+
+    @property
+    def inverse(self):
+        if self._inverse_func is not None:
+            return Transform(self._inverse_func, self.func)
+        else:
+            raise ValueError('Inverse function was not provided.')
+
+
+class ScaleTranslate(Transform):
+    """n-dimensional scale and translation (shift) class.
+
+    Scaling is always applied before translation.
+
+    Parameters
+    ----------
+    scale : 1-D array
+        A 1-D array of factors to scale each axis by. Scale is broadcast to 1
+        in leading dimensions, so that, for example, a scale of [4, 18, 34] in
+        3D can be used as a scale of [1, 4, 18, 34] in 4D without modification.
+        An empty translation vector implies no scaling.
+    translate : 1-D array
+        A 1-D array of factors to shift each axis by. Translation is broadcast
+        to 0 in leading dimensions, so that, for example, a translation of
+        [4, 18, 34] in 3D can be used as a translation of [0, 4, 18, 34] in 4D
+        without modification. An empty translation vector implies no
+        translation.
+    name : string
+        A string name for the transform.
+    """
+
+    def __init__(self, scale=(1.0,), translate=(0.0,), name=None):
+        super().__init__(name=name)
+        self.scale = np.array(scale)
+        self.translate = np.array(translate)
+
+    def __call__(self, coords):
+        coords = np.atleast_2d(coords)
+        scale = np.concatenate(
+            ([1.0] * (coords.shape[1] - len(self.scale)), self.scale)
+        )
+        translate = np.concatenate(
+            ([0.0] * (coords.shape[1] - len(self.translate)), self.translate)
+        )
+        return np.squeeze(scale * coords + translate)
+
+    @property
+    def inverse(self):
+        """Return the inverse transform."""
+        return ScaleTranslate(1 / self.scale, -1 / self.scale * self.translate)
+
+    def set_slice(self, axes: Sequence[int]):
+        """Return a transform subset to the visible dimensions."""
+        return ScaleTranslate(self.scale[axes], self.translate[axes])
+
+    def set_pad(self, axes: Sequence[int]):
+        """Return a transform with added axes for non-visible dimensions."""
```

Comment:
Can you add Parameters and Returns sections? It's unclear from this signature and docstring what the correct input for this method is. I would have thought that `set_pad` would take in an integer, the number of total axes.

Reply:
Done

Reply:
Note `set_pad` takes a list so that it can act like an inverse to `set_slice` and pad not just at the beginning or end, but at desired locations